device/workstation/w_windows.go

```diff
@@ -1,11 +1,268 @@
 package workstation
 
-// @todo add windows methods
+import (
+	"fmt"
+	"io"
+	"os/exec"
+
+	"math"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"encoding/csv"
+	"github.com/xshellinc/tools/dialogs"
+	"github.com/xshellinc/tools/lib/help"
+)
+
+var cleanTemplate = `
+select disk %s
+clean
+create partition primary
+active
+assign letter=N
+remove letter=N
+format fs=fat32 label=New quick
+`
 
 type windows struct {
 	*workstation
+	ddPath string
 }
 
 func newWorkstation() WorkStation {
+	m := new(MountInfo)
+	var ms []*MountInfo
+	return &windows{&workstation{runtime.GOOS, true, m, ms}, ""}
+}
+
+// Lists available mounts
+func (w *windows) ListRemovableDisk() error {
+	log.Debug("Listing disks...")
+	var out = []*MountInfo{}
+
+	stdout, err := help.ExecCmd("wmic", []string{"diskdrive", "get", "DeviceID,index,InterfaceType,MediaType,Model,Size", "/format:csv"})
+	log.Debug(stdout)
+	if err != nil {
+		stdout = ""
+	}
+	r := csv.NewReader(strings.NewReader(strings.TrimSpace(stdout)))
+	r.TrimLeadingSpace = true
+	r.Read() //skip the first line
+	for {
+		if record, err := r.Read(); err == io.EOF {
+			break
+		} else if err == nil {
+			if !strings.Contains(record[4], "Removable") || strings.Contains(record[3], "IDE") {
+				continue
+			}
+			var p = &MountInfo{}
+			size := record[6]
+			p.deviceSize = size
+			sizeInt, _ := strconv.Atoi(size)
+			sizeFloat := math.Ceil(float64(sizeInt) / 1024 / 1024 / 1024)
+			p.deviceName = record[5] + " [" + strconv.Itoa(int(sizeFloat)) + "GB]"
+			p.diskName = `\\?\Device\Harddisk` + record[2] + `\Partition0`
+			p.diskNameRaw = record[2]
+			out = append(out, p)
+		}
+	}
+	log.WithField("out", out).Debug("got drives")
+	if !(len(out) > 0) {
+		return fmt.Errorf("[-] No removable disks found, please insert your SD card and try again.\n[-] Please remember to run this tool as an administrator.")
+	}
+	w.workstation.mounts = out
 	return nil
 }
+
+// Unmounts the disk
+func (w *windows) Unmount() error {
+	return nil
+}
+
+// Ejects the mounted disk
+func (w *windows) Eject() error {
+	return nil
+}
+
+const diskSelectionTries = 3
+const writeAttempts = 5
+
+// Notifies user to choose a mount, after that it tries to write the data with `diskSelectionTries` number of retries
+func (w *windows) WriteToDisk(img string) (job *help.BackgroundJob, err error) {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 21 of function (*windows).WriteToDisk() is high (> 15)

<!-- Comment by [SideCI](https://sideci.com) -->